docs/develop/typescript-restrictions.md

```diff
@@ -76,6 +76,31 @@ let filteredArray = myArray.filter((x) => {
 */
 ```
 
+## Unions of `ExcelScript` types and user-defined types aren't supported
+
+Office Scripts are converted at runtime from synchronous to asynchronous code blocks. The communication with the workbook through [Promises](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) is hidden from the script creator. This conversion doesn't support [union types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) that include `ExcelScript` types and user-defined types. In that case, the `Promise` is returned to the script, but the Office Script complier doesn't expect it and the script creator can't interact with the `Promise`.
```

Comment:
Two changes: 
1. We decided in team meeting on 3/16 that "promise" should only be capitalized when wrapped in code ticks, because Term Studio identifies "promise" as a regular noun and not a proper noun, and MSFT style guide recommends against capitalization unless strictly necessary.
2. Typo fix
```suggestion
Office Scripts are converted at runtime from synchronous to asynchronous code blocks. The communication with the workbook through [promises](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) is hidden from the script creator. This conversion doesn't support [union types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) that include `ExcelScript` types and user-defined types. In that case, the `Promise` is returned to the script, but the Office Script compiler doesn't expect it and the script creator can't interact with the `Promise`.
```